src/include/gurt/atomic.h

```diff
@@ -0,0 +1,83 @@
+/* 
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+/*
+ * Atomic directives
+ */
+#ifndef __GURT_ATOMIC_H__
+#define __GURT_ATOMIC_H__
+
+#if HAVE_STDATOMIC
+
+#include <stdatomic.h>
+
+#ifdef __INTEL_COMPILER
+#define ATOMIC volatile
```

Comment:
(style)  Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt